xarray/backends/api.py

```diff
@@ -743,19 +748,23 @@ def save_mfdataset(datasets, paths, mode='w', format=None, groups=None,
                          'save_mfdataset')
 
     writer = ArrayWriter()
-    stores = [to_netcdf(ds, path, mode, format, group, engine, writer)
+    stores = [to_netcdf(ds, path, mode, format, group, engine, writer,
+                        compute=compute)
               for ds, path, group in zip(datasets, paths, groups)]
+
     try:
-        writer.sync()
+        delayed = writer.sync(compute=compute)
         for store in stores:
-            store.sync()
+            store.sync(compute=compute)
```

Comment:
I'm not sure why/if we need this sync line here. My understanding is that syncing the writer should be enough here.

Reply:
Sometimes you need to sync a backend object to actually save data to disk. Synchronizing the writer only saves data to the backend's arrays, but that data is sometimes queued up before writing (e.g., with scipy, which only writes full netCDF files).